[Jenkins] Unify Windows Headers and Unit tests to cut time required

seantalts opened the pull request:
On windows, setup and teardown is particularly expensive, taking something like 3 or 4 minutes. This quick PR just does Windows Unit tests immediately after the Windows Header tests in the same directory rather than deleting the whole directory and unstashing in between.

    * Copyright holder: Columbia University

syclik commented:
@seantalts, this looks fine, but any ideas why it failed? (Did that Windows machine get an upgrade?)

seantalts commented:
Yep, looks like something I'll have to dig into.
On Thu, Feb 14, 2019 at 20:23 Daniel Lee <notifications@github.com> wrote:

> @seantalts <https://github.com/seantalts>, this looks fine, but any ideas
> why it failed? (Did that Windows machine get an upgrade?)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/stan-dev/math/pull/1119#issuecomment-463870010>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAxJ7OfJpV-HZo5Y8wkhPqC6YRTlWK-Uks5vNgv5gaJpZM4a8WSt>
> .
>
-- 
📲

syclik commented:
Thanks.

On Thu, Feb 14, 2019 at 9:18 PM seantalts <notifications@github.com> wrote:

> Yep, looks like something I'll have to dig into.
> On Thu, Feb 14, 2019 at 20:23 Daniel Lee <notifications@github.com> wrote:
>
> > @seantalts <https://github.com/seantalts>, this looks fine, but any
> ideas
> > why it failed? (Did that Windows machine get an upgrade?)
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/stan-dev/math/pull/1119#issuecomment-463870010>, or
> mute
> > the thread
> > <
> https://github.com/notifications/unsubscribe-auth/AAxJ7OfJpV-HZo5Y8wkhPqC6YRTlWK-Uks5vNgv5gaJpZM4a8WSt
> >
> > .
> >
> --
> 📲
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/stan-dev/math/pull/1119#issuecomment-463880853>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAZ_F7yJjbBJhsJVDAcyOITmz4N8fJ33ks5vNhkLgaJpZM4a8WSt>
> .
>

seantalts commented:
Fixed! Merging...

syclik commented:
Cool. Know what caused it?

On Fri, Feb 15, 2019 at 7:04 AM seantalts <notifications@github.com> wrote:

> Fixed! Merging...
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/stan-dev/math/pull/1119#issuecomment-464024041>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAZ_F9yoDEGLlTEv0oqlMIKpfet30NWEks5vNqJagaJpZM4a8WSt>
> .
>

seantalts commented:
to install WSL on the machine after the upgrade, we created a new non-admin Jenkins user that we switched the Jenkins agent to run as, and it didn't have permission to everything in the workspace. So I cleaned up the workspace, deletling ~80GB of old jobs, and set the permissions on the folder to allow the new Jenkins user full control. Apparently it had been running as the local system account :P

syclik commented:
Sweet! Thanks for hunting it down.

On Fri, Feb 15, 2019 at 8:40 AM seantalts <notifications@github.com> wrote:

> Merged #1119 <https://github.com/stan-dev/math/pull/1119> into develop.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/stan-dev/math/pull/1119#event-2142501726>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAZ_F_S4cSK7B5-AE6FCDD7eppmQNrkBks5vNrjPgaJpZM4a8WSt>
> .
>
